TrackletAlgorithm/MatchCalculatorTop.cpp

```diff
@@ -18,7 +18,8 @@ void MatchCalculatorTop(BXType bx,
                         FullMatchMemory<BARREL>* fullmatch4,
                         FullMatchMemory<BARREL>* fullmatch5,
                         FullMatchMemory<BARREL>* fullmatch6,
-                        FullMatchMemory<BARREL>* fullmatch7
+                        FullMatchMemory<BARREL>* fullmatch7,
+                        FullMatchMemory<BARREL>* fullmatch8
                         )
 {
  #pragma HLS inline off
```

Comment:
This hasn't changed, but I'm wondering why inlining is turned off here. I believe we found that in these wrapper functions, HLS wastes a clock cycle with the extra function call unless inlining is turned on.

Reply:
I found the same thing with projection router. We should explicitly inline the call to the templated function in the top-level function, otherwise we waste a clock. I'll make these changes.

Reply:
Actually I misspoke (or miscommented). Inlining the Top-level wrapper function has no effect because the inline pragma applies upward to functions that call it (which is none, in the top-level). The inline pragma I think needs to be in the templated generic function (although I think by default they are inlined anyways). This is my experience, @aehart do you remember if that's what you saw as well?

Reply:
Oh right. I think I had that backwards because I use the `recursive` option in the TC in order to satisfy pipelining requirements and avoid any latency from the function hierarchy there. So you're right that the pragma should be in the generic template function, and in this example, I guess turning inlining off has no effect here.

I don't remember seeing the inlining happen by default, but it should be easy to check if it happens without the pragma. Even if it does though, it might be worth adding the pragma just to be explicit.

Reply:
Agreed. I'll add the pragmas and push momentarily